server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v67/MigratePreviousAnalysisLeakPeriodTest.java

```diff
@@ -0,0 +1,188 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.db.CoreDbTester;
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphanumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MigratePreviousAnalysisLeakPeriodTest {
+
+  private final static String SELECT_PROPERTIES = "SELECT prop_key, is_empty, text_value, clob_value FROM properties";
+
+  @Rule
+  public CoreDbTester db = CoreDbTester.createForSchema(MigratePreviousAnalysisLeakPeriodTest.class, "properties.sql");
+
+  private MigratePreviousAnalysisLeakPeriod underTest = new MigratePreviousAnalysisLeakPeriod(db.database());
+
+  @Test
+  public void migration_must_update_the_database() throws SQLException {
+    List<Property> properties = generateAndInsertProperties();
+
+    underTest.execute();
+
+    assertDatabaseContainsExactly(applyChanges(properties));
+  }
+
+  @Test
+  public void migration_must_be_reentrant() throws SQLException {
+    List<Property> properties = generateAndInsertProperties();
+
+    underTest.execute();
+    underTest.execute();
+
+    assertDatabaseContainsExactly(applyChanges(properties));
+  }
+
+  @Test
+  public void migration_is_doing_nothing_when_no_data() throws SQLException {
+    assertThat(db.countRowsOfTable("properties")).isEqualTo(0);
+    underTest.execute();
+    assertThat(db.countRowsOfTable("properties")).isEqualTo(0);
+  }
+
+  @Test
+  public void update_must_look_in_clob() throws SQLException {
+    Property property = new Property("sonar.leak.period", null,"previous_analysis", false);
+    property.insert();
+
+    underTest.execute();
+
+    assertThat(db.select(SELECT_PROPERTIES)).isEqualTo(
+      ImmutableList.of(new Property("sonar.leak.period", "previous_version",null, false).toMap())
+    );
+  }
+
+  @Test
+  public void update_must_look_in_text_value() throws SQLException {
+    Property property = new Property("sonar.leak.period", "previous_analysis", null,false);
+    property.insert();
+
+    underTest.execute();
+
+    assertThat(db.select(SELECT_PROPERTIES)).isEqualTo(
+      ImmutableList.of(new Property("sonar.leak.period", "previous_version",null, false).toMap())
+    );
+  }
+
+  private void assertDatabaseContainsExactly(List<Property> expectedProperties) {
+    assertThat(db.select(SELECT_PROPERTIES)).isEqualTo(
+      expectedProperties.stream().map(Property::toMap).collect(Collectors.toList())
+    );
+  }
+
+  private List<Property> generateAndInsertProperties() {
+    List<Property> properties = ImmutableList.of(
+      new Property("sonar.leak.period", randomAlphanumeric(1000), null, false),
+      new Property("sonar.leak.period", "previous_version", null, false),
+      new Property("sonar.leak.period", null, "previous_analysis", false),
```

Comment:
This use case does not exist => clob_value will never contains "previous_analysis"

Reply:
It does not exist currently but I was not sure that it was not the case before. I'll remove it.